src/v2/Components/UserSettings/TwoFactorAuthentication/helpers.ts

```diff
@@ -0,0 +1,17 @@
+import { getClientParam } from "v2/Utils/getClientParam"
+import sanitizeRedirect from "@artsy/passport/sanitize-redirect"
+
+export const afterUpdateRedir = () => {
+  const afterUpdateURL = getClientParam("after_update")
+
+  if (afterUpdateURL) {
+    const sanitizedURL = sanitizeRedirect(afterUpdateURL)
+
+    // a '/' means URL is bad
```

Comment:
Surprising... what if a caller was trying to redirect to the root page?

Reply:
not sure. shall test.

Reply:
tested. if `after_update` points to `/`:

`http://localhost:5000/user/edit?after_update=%2F#two-factor-authentication`

[sanitize_redirect](https://github.com/artsy/force/blob/master/src/mobile/components/sanitize_redirect/index.coffee) returns `/` which causes `afterUpdateRedirect` to return empty string, so the old `Log back in` modal is displayed. I think this is desired behavior?

if `after_update` is a relative path other than `/`, say `/foo`:

`http://localhost:5000/user/edit?after_update=%2Ffoo#two-factor-authentication`

sanitize_redirect returns `/foo` which does cause the redirect modal to show which displays no target hostname (b/c `/foo` is relative), and clicking OK takes one to `localhost:5000/foo` and 404's. Ooo, that's bad.

i guess we should show the redirect modal only when `after_update` points to an absolute URL?

Reply:
This case is totally fine with the special `/` meaning. I was just surprised it operated that way generally for artsy-passport's other cases.

Ideally the new `after_update` param would accept either absolute or relative URLs, but you're right that the modal's copy would need to change in that case. In the case of relative redirects, I'd be fine if it just didn't say the "you'll be redirected..." bit.

Reply:
revised. now, the presence of `after_update` param always cause the redirect modal to show. if the redir url is absolute, the modal shows `You will be redirected to: <host>`. if the url is relative (`/`, `/foo`, ...), the modal omits `You will be redirected...` and it will only say `You’ve successfully set up two-factor authentication!` and clicking OK will still redirect to that relative url.